.github/workflows/release.yml

```diff
@@ -4,46 +4,197 @@ on:
+        run: git fetch --force origin $(git describe --tags):refs/tags/$(git describe --tags)
+
+      - name: Get version number
+        run: git describe | tee version
+
+      - name: Extract tag body
+        run: git tag --points-at "${GITHUB_REF}" --format '%(contents:body)' | grep . | tee release-body
+
+      - name: Bootstrap automake
+        run: ./autogen.sh
+
+      - name: XXX JUMPSTART
+        run: GITHUB_BASE=cockpit-project/cockpit tools/webpack-jumpstart --partial
+
+      - name: Build release
+        run: make dist -j$(nproc)
+
+      - name: Note source archive name
+        run: basename "$(make dump-dist)" | tee source-filename
+
+      - id: publish
+        name: Publish release
+        uses: actions/github-script@v4
+        with:
+          script: |
+            const fs = require('fs');
+            const version = fs.readFileSync('version', {encoding: 'ascii'}).trim();
+            const source_filename = fs.readFileSync('source-filename', {encoding: 'ascii'}).trim();
+            const release_body = fs.readFileSync('release-body', {encoding: 'utf-8'});
+
+            const source = fs.readFileSync(source_filename);
+
+            const release = await github.repos.createRelease({
+                owner: context.repo.owner, repo: context.repo.repo,
+                tag_name: version, name: version, body: release_body, draft: true
+                });
+
+            /* We get a lot of connection reset errors... */
+            for (let retry = 0; retry < 5; retry++) {
+                try {
+                    const upload = await github.repos.uploadReleaseAsset({
+                        owner: context.repo.owner, repo: context.repo.repo, release_id: release.data.id,
+                        name: source_filename, headers: {'Content-Type': 'application/x-xz'}, data: source});
+                    break;
+                } catch (error) {
+                    console.log(`Failed to upload (status ${error.status})`);
+                    if (error.status != 500) {
+                        throw error;
+                    }
+                }
+            }
+
+            const update = await github.repos.updateRelease({
+              owner: context.repo.owner, repo: context.repo.repo, release_id: release.data.id,
+              draft: false});
+
+            core.setOutput('checksum', require('crypto').createHash('sha256').update(source).digest('hex'));
+            core.setOutput('download', update.data.assets[0].browser_download_url);
+            core.setOutput('filename', source_filename);
+            core.setOutput('version', version);
+
+    outputs:
+      checksum: ${{ steps.publish.outputs.checksum }}
+      download: ${{ steps.publish.outputs.download }}
+      filename: ${{ steps.publish.outputs.filename }}
+      version: ${{ steps.publish.outputs.version }}
+
+
+  guide:
+    needs: source
+    environment: website
+    permissions:
+      pull-requests: write
```

Comment:
Why? This directly commits the update, no PR involved. This job does not even use `secrets.GITHUB_TOKEN`. As this uses a deploy key, `permissions: {}` ought to work?

Reply:
Done.